pkg/model/route.go

```diff
@@ -0,0 +1,64 @@
+//
+// Copyright (c) 2023 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	openshift "github.com/openshift/api/route/v1"
+	bsv1alpha1 "janus-idp.io/backstage-operator/api/v1alpha1"
+	"janus-idp.io/backstage-operator/pkg/utils"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type BackstageRouteFactory struct{}
+
+func (f BackstageRouteFactory) newBackstageObject() BackstageObject {
+	return &BackstageRoute{route: &openshift.Route{}}
+}
+
+type BackstageRoute struct {
+	route *openshift.Route
+}
+
+func init() {
+	registerConfig("route.yaml", BackstageRouteFactory{}, ForOpenshift)
+}
+
+// implementation of BackstageObject interface
+func (b *BackstageRoute) Object() client.Object {
+	return b.route
+}
+
+// implementation of BackstageObject interface
+func (b *BackstageRoute) EmptyObject() client.Object {
+	return &openshift.Route{}
+}
+
+// implementation of BackstageObject interface
```

Comment:
Why the route related features are dropped?
https://github.com/janus-idp/operator/pull/94
https://github.com/janus-idp/operator/pull/79

Refactoring should not remove existing capabilities.

Reply:
It is not dropped, tests passed and as mentioned https://github.com/janus-idp/operator/pull/70#issuecomment-1887454951 "Route is now working properly with the latest changes"

Reply:
@gazarenkov The route only works in the default configuration. I tested the following CR and found that:
1) support for route.enabled is missing.
2) Support for route.host/subdomain does not work.
3)Support for route.tls certificates in the route does not work.
So basically all functionalities implemented in the previous PRs are dropped. 
https://github.com/janus-idp/operator/pull/94
https://github.com/janus-idp/operator/pull/79

```
apiVersion: janus-idp.io/v1alpha1
kind: Backstage
metadata:
  name: bs1
spec:
  database:
    enableLocalDb: true
    authSecretName: test-postgres-secrets
  application:
    imagePullSecrets:
    - my-pull-secret
    replicas: 2
    route:
      enabled: false
      host: test.com
      subdomain: mysubdomain
    extraEnvs:
      envs:
        - name: myenv1
          value: my-value-1
        - name: myenv2
          value: my-value-2
```